### ISSUE TITLE
Huge volume of logs and number of stack traces

### ISSUE THREAD

**Issue by jakubgs** [status-im/status-go#3543]
### Problem

The volume of logs from the `tests` CI job is ridiculous:
```
 > wc -l status_go_tests.log
258778 status_go_tests.log

 > du -hs status_go_tests.log
49M	status_go_tests.log
```
It's almost not readable in the browser, and `vim` requires a `-u NONE` or `'+no syntax'` allow the file to load.

### Future Steps

One thing that would be great to do would be to split the logs from tests into separate files depending on test.
In a perfect world we could also prefix those log files with `failure_` or `success_` to easily find the issue.
This might be very difficult or impossible to do with Go test logging, so we should try to reduce the volume first.

**Comment by jakubgs** [comment]
One weird thing I've noticed is that a noticable portion of the logs are actually stack traces:
```
 > grep -e 'Z] github.com/status-im/status-go' -e 'Z]   /home/jenkins/workspace/status-go' status_go_tests.log | wc -l
21665
```

**Comment by jakubgs** [comment]
They appear after warning log level messages for some reason, in effect making the log file very hard to follow:
```
 > grep -B1 -e 'Z] github.com/status-im/status-go' -e 'Z]       /home/jenkins/workspace/status-go' status_go_tests.log | tail -n5
2023-05-29T10:46:54.162Z	WARN	wakuv2/waku.go:440	invalid peer multiaddress	{"addr": "/ip4/47.242.202.59/tcp/30303", "error": "invalid p2p multiaddr"}
github.com/status-im/status-go/wakuv2.(*Waku).identifyAndConnect
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/wakuv2/waku.go:440
github.com/status-im/status-go/wakuv2.(*Waku).addWakuV2Peers.func1.1
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/wakuv2/waku.go:402
```
```
2023-05-29T10:55:50.728Z	WARN	protocol/messenger_mailserver_cycle.go:65	not starting mailserver cycle
github.com/status-im/status-go/protocol.(*Messenger).StartMailserverCycle
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/messenger_mailserver_cycle.go:65
github.com/status-im/status-go/protocol.(*Messenger).Start
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/messenger.go:743
github.com/status-im/status-go/protocol.newMessengerWithKey
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/messenger_base_test.go:118
github.com/status-im/status-go/protocol.(*MessengerPushNotificationSuite).newMessenger
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/push_notification_test.go:70
github.com/status-im/status-go/protocol.(*MessengerPushNotificationSuite).TestReceivePushNotificationRetries
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/push_notification_test.go:429
```
And some errors:
```
2023-05-29T10:55:55.059Z	ERROR	pushnotificationclient/client.go:888	could not retrieve message	{"error": "sql: database is closed"}
github.com/status-im/status-go/protocol/pushnotificationclient.(*Client).handlePublicMessageSent
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/pushnotificationclient/client.go:888
github.com/status-im/status-go/protocol/pushnotificationclient.(*Client).handleMessageSent
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/pushnotificationclient/client.go:842
github.com/status-im/status-go/protocol/pushnotificationclient.(*Client).subscribeForMessageEvents.func1
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/pushnotificationclient/client.go:718
```
```
2023-05-29T10:46:48.607Z	ERROR	pushnotificationserver/server.go:293	failed to verify grant	{"error": "invalid signature length"}
github.com/status-im/status-go/protocol/pushnotificationserver.(*Server).validateRegistration
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/pushnotificationserver/server.go:293
github.com/status-im/status-go/protocol/pushnotificationserver.(*ServerSuite).TestPushNotificationServerValidateRegistration
	/home/jenkins/workspace/status-go_prs_tests_PR-3542/protocol/pushnotificationserver/server_test.go:264
```

**Comment by jakubgs** [comment]
While working on these PRs:

* https://github.com/status-im/status-go/pull/3663
* https://github.com/status-im/status-go/pull/3669

I realized we could make use of `go test -json` and generate nice HTML or JUnit reports using one of these tools:

* https://github.com/vakenbolt/go-test-report
* https://github.com/jstemmer/go-junit-report

And then provide those to Jenkins for easier consumption by devs.

**Comment by yakimant** [comment]
JUnit Jenkins integration:
https://www.jenkins.io/doc/pipeline/steps/junit/

# code coverage format - if we'll want it in the future

gocover to cobertura conversion tools:
- https://github.com/t-yuki/gocover-cobertura
- https://github.com/AlekSi/gocov-xml

cobertura plugin for Jenkins:
- https://plugins.jenkins.io/cobertura/
- https://www.jenkins.io/doc/pipeline/steps/cobertura/

# html publishing in jenkins
https://plugins.jenkins.io/htmlpublisher/

**Comment by yakimant** [comment]
Unit tests integration to jenkins is done, but it doesn't solve an issue with large `status-go` logs.
They are the largest on CI, take a lot of space, difficult to download, read, etc.

**Comment by yakimant** [comment]
DEBUG logs take 95k, stacktrace 34k and INFO 16k out of 152k.

@richard-ramos, @andremedeiros, @iurimatias, are they useful for test failure investigation locally or on CI?

<details>
<summary>See more details on log stats</summary>
<pre>
❯ cat log\(18\).txt | wc -l
  152163

❯ cat log\(18\).txt| grep DEBUG | wc -l
   95141

❯ cat log\(18\).txt|  grep -e 'Z]\sreflect' -e 'Z] github.com' -e 'Z]\s*/home/jenkins/workspace/status-go' -e 'Z]\s*/nix/store/' -e 'Z]\s*testing' | wc -l
   33760

❯ cat log\(18\).txt| grep INFO | wc -l
   15636

❯ cat log\(18\).txt|  grep -e 'Z]\svendoring' -e 'Z]\sgo:' | wc -l
    2693

❯ cat log\(18\).txt| grep WARN | wc -l
    1889

❯ cat log\(18\).txt| grep ERROR | wc -l
    1670

❯ cat log\(18\).txt| grep -v DEBUG | grep -v WARN | grep -v INFO | grep -v ERROR | grep -v -e 'Z]\s*reflect' -e 'Z] github.com' -e 'Z]\s*/home/jenkins/workspace/status-go' -e 'Z]\s*/nix/store/' -e 'Z]\s*testing' -e 'Z]\svendoring' -e 'Z]\sgo:' | wc -l
    1374
</pre>
</details>

**Comment by jakubgs** [comment]
Good work, thanks.
